src/AutoMapper/TypeMap.cs

```diff
@@ -328,4 +338,39 @@ private void ApplyInheritedTypeMap(TypeMap inheritedTypeMap)
             }
         }
     }
+
+    public class TypeMap<TSource, TDestination>

```

Comment:
Why do we have to go through the TypeMap class again? Instead of inlining the expressions? Is it something CompileToMethod requires?